src/jit/optimizer.cpp

```diff
@@ -8410,7 +8410,7 @@ bool Compiler::optExtractArrIndex(GenTree* tree, ArrIndex* result, unsigned lhsN
     {
         return false;
     }
-#ifdef _TARGET_AMD64_ // TODO-ARM64: should this be _TARGET_64BIT_? If not, add comment why not.
+#ifdef _TARGET_64BIT_
     if (index->gtOper != GT_CAST)
```

Comment:
What cast do we expect here for `_TARGET_64BIT_`?

`#else // !_TARGET_64BIT_`

Reply:
Casts from int to long on 64 bit platforms.